code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -100,17 +100,17 @@ emp_act
 
 
 /mob/living/carbon/human/proc/check_shields(var/damage = 0, var/attack_text = "the attack")
-	if(l_hand && istype(l_hand, /obj/item/weapon))
+	if(istype(l_hand, /obj/item/weapon)) //Check left hand
```

Comment:
Why are you removing the check? It's too prevent runtimes if the human doesn't have anything in that slot.